test/e2e/prometheus_test.go

```diff
@@ -84,3 +85,98 @@ func TestPrometheusVolumeClaim(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestPrometheusAlertmanagerAntiAffinity(t *testing.T) {
+	pods, err := f.KubeClient.CoreV1().Pods(f.Ns).List(metav1.ListOptions{FieldSelector: "status.phase=Running"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var alm = `affinity:
+	   podAntiAffinity:
+	      preferredDuringSchedulingIgnoredDuringExecution:
+	      - podAffinityTerm:
+	          labelSelector:
+	            matchExpressions:
+	            - key: alertmanager
+	              operator: In
+	              values:
+	              - main`
+
+	var k8s = `affinity:
+	   podAntiAffinity:
+	      preferredDuringSchedulingIgnoredDuringExecution:
+	      - podAffinityTerm:
+	          labelSelector:
+	            matchExpressions:
+	            - key: prometheus
+	              operator: In
+	              values:
+	              - k8s`
+
+	var testPod1 = "alertmanager-main"
+	var testPod2 = "prometheus-k8s"
+	var testNameSpace = "openshift-monitoring"
+
+	mapPodAntiAff := map[int]string{0: ""}
+	for idx, line := range strings.Split(alm, "\n") {
+		mapPodAntiAff[idx] = line
+	}
+	mapPodK8s := map[int]string{0: ""}
+	for idx, line := range strings.Split(k8s, "\n") {
+		mapPodK8s[idx] = line
+	}
+	
+	var resultCheckAlertmanagerMain = false
+	var resultCheckPrometheuseK8s = false
+	
+	for _, p := range pods.Items {
+		if strings.Contains(p.Namespace, testNameSpace) &&
+			strings.Contains(p.Name, testPod1) {
+			outputPodAntiAffinity := p.Spec.Affinity.PodAntiAffinity.String()
+			podAntiAffinity := strings.ToLower(strings.Split(
+				strings.TrimSpace(mapPodAntiAff[1]), ":")[0])
+			preferredDuringSchedulingIgnoredDuringExecution := strings.ToLower(strings.Split(
+				strings.TrimSpace(mapPodAntiAff[2]), ":")[0])
+			last := len(mapPodAntiAff)
+			values := strings.ToLower(strings.Split(
+				strings.TrimSpace(mapPodAntiAff[last-1]), " ")[1])
+			key := strings.ToLower(strings.Split(
+				strings.TrimSpace(mapPodAntiAff[6]), ":")[1])
+				if strings.Contains(strings.ToLower(outputPodAntiAffinity), podAntiAffinity) &&
+					strings.Contains(strings.ToLower(outputPodAntiAffinity), preferredDuringSchedulingIgnoredDuringExecution) && 
+					strings.Contains(strings.ToLower(outputPodAntiAffinity), values) &&
+					strings.Contains(strings.ToLower(outputPodAntiAffinity), strings.TrimSpace(key)) {
+					resultCheckAlertmanagerMain = true
+					} else {
```

Comment:
please fix indentation according to `gofmt`.

Reply:
fixed